src/main/java/org/red5/server/net/rtmp/RTMPConnection.java

```diff
@@ -644,7 +644,7 @@ public Number reserveStreamId() {
 			}
 		}
 
-		if (d == MAX_RESERVED_STREAMS) {
+		if ((int) d == MAX_RESERVED_STREAMS) {

```

Comment:
If d is a Number use Number.intValue() in this type of situation, but I think a better fix is to change the type of MAX_RESERVED_STREAMS to a double instead.